### PR TITLE
Krzysztof room info import

### PR DESCRIPTION
**Pull request by KrzBan**
RoomImportService + RoomService

### CODE REVIEW

File: backend/src/main/java/com/wpz/rbs/service/RoomService.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        roomRepository.findAll().forEach(room -> rooms.add(room));  
+        return rooms;  
+    }  
+ 
+    public Room getById(int id){  
+        return roomRepository.findById(id).get();  
+    }  
+
+    // When updating, keeps the "adnotation" member, unless it's empty ("")
+    public void saveOrUpdate(Room example){
```
Comment (bestemic) [inline backend/src/main/java/com/wpz/rbs/service/RoomService.java:29]: rename example to room

File: backend/src/main/java/com/wpz/rbs/service/RoomService.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            var existingAdnotation = optional.get().getAdnotation();
+
+            if(existingAdnotation.equals("")){
+                example.setAdnotation(existingAdnotation);
+            }
+        }
+        roomRepository.save(example);  
+    }  
+
+    public void delete(int id){
```
Comment (bestemic) [inline backend/src/main/java/com/wpz/rbs/service/RoomService.java:41]: we don't want to delete room

File: backend/src/main/java/com/wpz/rbs/controller/RoomController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        return roomService.getAll();
+    }
+
+    @GetMapping("/room/{id}")
+    private Room get(@PathVariable("id") int id){
+        return roomService.getById(id);
+    }
+
+    @DeleteMapping("/room/{id}")  
+    private void delete(@PathVariable("id") int id){
```
Comment (bestemic) [inline backend/src/main/java/com/wpz/rbs/controller/RoomController.java:33]: we don't want to delete room

File: backend/src/main/java/com/wpz/rbs/controller/RoomController.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        return roomService.getById(id);
+    }
+
+    @DeleteMapping("/room/{id}")  
+    private void delete(@PathVariable("id") int id){  
+        roomService.delete(id);  
+    }  
+
+    @PostMapping("/room")  
+    private int save(@RequestBody Room room){
```
Comment (bestemic) [inline backend/src/main/java/com/wpz/rbs/controller/RoomController.java:38]: we won't be creating our own rooms, we will be only updating information about existing one, PUT will be better

File: backend/src/main/java/com/wpz/rbs/service/RoomService.java [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        return roomRepository.findById(id).get();  
+    }  
+
+    // When updating, keeps the "adnotation" member, unless it's empty ("")
+    public void saveOrUpdate(Room example){  
+        var optional = roomRepository.findById(example.getId());
+        if(optional.isPresent()){
+            var existingAdnotation = optional.get().getAdnotation();
+
+            if(existingAdnotation.equals("")){
```
Comment (bestemic) [inline backend/src/main/java/com/wpz/rbs/service/RoomService.java:34]: when we will be want to update our annotaton this part will won't work

**Review by bestemic — CHANGES_REQUESTED** [review]
Need some small impovements

File: backend/src/main/java/com/wpz/rbs/service/RoomService.java (hunk 5 above)
Comment (KrzBan) [reply backend/src/main/java/com/wpz/rbs/service/RoomService.java:34]: We present the default state, the empty string, as valid. With that in mind, after setting the annotation, it should be possible to return to such state, that being, the empty string. Just updating the annotation is going to require another function to be added, but topic as a whole should be discussed.

**Review by bestemic — APPROVED** [review]

**Pull request merged**